chore(linear_algebra/linear_independent): make a binding explicit in ne_zero

Julian opened the pull request:
Suggested by @PatrickMassot in
https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/independent.20columns/near/220700702

bryangingechen commented on src/linear_algebra/linear_independent.lean:
```diff
@@ -462,7 +462,7 @@ begin
         rw hxy,
         exact (subset_span (mem_range_self y₂)) },
-      exact false.elim ((hindep x₁).ne_zero h0) } },
+      exact false.elim (linear_independent.ne_zero _ (hindep x₁) h0) } },
```
We can still use dot notation, we just need to add an extra underscore:
```suggestion
      exact false.elim ((hindep x₁).ne_zero _ h0).elim } },
```

This should work for the changes in the other file as well.

Julian replied:
Interesting, I didn't realize that still worked I thought it'd look for `ne_zero` on `i`. Will give it a shot and update.

bryangingechen replied:
Yep, Lean looks for the first explicit argument with a type equal to the namespace, see [Section 9.1 in TPiL](https://leanprover.github.io/theorem_proving_in_lean/structures_and_records.html#declaring-structures).

Julian replied:
Addressed I believe. Thanks for the tip!

Julian committed:
chore(linear_algebra/linear_independent): make a binding explicit in ne_zero

Suggested by @PatrickMassot in
https://leanprover.zulipchat.com/#narrow/stream/113489-new-members/topic/independent.20columns/near/220700702

PatrickMassot commented on src/linear_algebra/affine_space/independent.lean:
```diff
@@ -219,7 +219,7 @@ begin
-  refine ha.ne_zero _,
+  refine ha.ne_zero _ _,
   { exact ⟨i, hij'⟩ },
   { exact vsub_eq_zero_iff_eq.mpr hij },
```
```suggestion
  exact ha.ne_zero ⟨i, hij'⟩ (vsub_eq_zero_iff_eq.mpr hij)
```

Julian replied:
Done, thanks.
